Tools/GNUMake/sites/Make.unknown

```diff
@@ -20,101 +20,34 @@ ifeq ($(USE_MPI),TRUE)
-
-    #
-    # mpich
-    #
-
-#    ifneq ($(BL_NO_FORT),TRUE)
-      mpif90_link_flags := $(shell $(MPI_OTHER_COMP) -link_info)
-      #
-      # The first word is the underlying compiler say gfortran
-      #
-      LIBRARIES += $(wordlist 2,1024,$(mpif90_link_flags))
-#    endif
-
-#    ifneq ($(CXX),$(filter $(CXX),mpicxx mpic++))
-      mpicxx_link_flags := $(shell mpicxx -link_info)
-      LIBRARIES += $(filter -l%,$(mpicxx_link_flags))
-#    endif
-
-  else ifneq ($(findstring mvapich,$(shell $(MPI_OTHER_COMP) -show 2>&1 | tr A-Z a-z)),)
-
-    #
-    # mvapich
-    #
-
-#    ifneq ($(BL_NO_FORT),TRUE)
-      mpif90_link_flags := $(shell $(MPI_OTHER_COMP) -link_info)
-      #
-      # The first word is the underlying compiler say gfortran
-      #
-      LIBRARIES += $(wordlist 2,1024,$(mpif90_link_flags))
-#    endif
-
-#    ifneq ($(CXX),$(filter $(CXX),mpicxx mpic++))
-      mpicxx_link_flags := $(shell mpicxx -link_info)
-      LIBRARIES += $(filter -l%,$(mpicxx_link_flags))
-#    endif
-
-  else ifneq ($(findstring Open MPI, $(shell $(MPI_OTHER_COMP) -showme:version 2>&1)),)
-
-    #
-    # openmpi
-    #
-
-#    ifneq ($(BL_NO_FORT),TRUE)
-      mpif90_link_flags := $(shell $(MPI_OTHER_COMP) -showme:link)
-      LIBRARIES += $(mpif90_link_flags)
-#    endif
-
-#    ifneq ($(CXX),$(filter $(CXX),mpicxx mpic++))
-      mpicxx_link_flags := $(shell mpicxx -showme:link)
-      LIBRARIES += $(filter -l%,$(mpicxx_link_flags))
-#    endif
-
-     DEFINES += -DOMPI_SKIP_MPICXX
-
-  else ifneq ($(findstring Spectrum MPI, $(shell $(MPI_OTHER_COMP) -showme:version 2>&1)),)
-
-    #
-    # Spectrum MPI
-    #
-#    ifneq ($(BL_NO_FORT),TRUE)
-      mpif90_link_flags := $(shell $(MPI_OTHER_COMP) -showme:link)
-      LIBRARIES += $(mpif90_link_flags)
-#    endif
-
-  else ifneq ($(findstring /intel/mpi-rt,$(shell $(MPI_OTHER_COMP) -show 2>&1)),)
-
-    #
-    # Intel mpi
-    #
-#
-#    ifneq ($(BL_NO_FORT),TRUE)
-      mpif90_link_flags := $(shell $(MPI_OTHER_COMP) -link_info)
-      #
-      # The first word is the underlying compiler say gfortran
-      #
-      LIBRARIES += $(wordlist 2,1024,$(mpif90_link_flags))
-#    endif
-
-#    ifneq ($(CXX),$(filter $(CXX),mpicxx mpic++))
-      mpicxx_link_flags := $(shell mpicxx -link_info)
-      LIBRARIES += $(filter -l%,$(mpicxx_link_flags))
-#    endif
-
+  # error codes - find flag that displays the data
+  code_sml := $(shell $(MPI_OTHER_COMP) -showme:link > /dev/null 2>&1; echo $$?)
+  code_li :=  $(shell $(MPI_OTHER_COMP) -link_info   > /dev/null 2>&1; echo $$?)
+
+  # f90: filter all flags (-%)
+  # cxx: filter additional library flags (-l%)
+  ifeq ($(code_sml), 0)
+     mpif90_link_flags := $(filter -%,$(shell $(MPI_OTHER_COMP) -showme:link))
+     mpicxx_link_flags := $(filter -l%,$(shell mpicxx -showme:link))
+  else ifeq ($(code_li), 0)
+     mpif90_link_flags := $(filter -%,$(shell $(MPI_OTHER_COMP) -link_info))
```

Comment:
I don't think it is safe to filter-in `-`.  We should instead filter out the first word if it doe not start with `-`.  On one machine I am testing `mpif90 -link_info` gives
```
gfortran .... -Xlinker '/glob/.....'.
``` 
Because of the space between `-Xlinker` and `/glob/...`, the latter gets dropped.  This could also happen to `-I xxx` and `-L xxx`.

Reply:
I was worried about that one. I believe "eliminate the first word if it doesn't start with a dash" should be pretty doable. I'll try writing that instead.